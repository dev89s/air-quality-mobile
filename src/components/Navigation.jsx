import { useEffect, useState } from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import backIcon from '../assets/back.png';

function Navigation() {
  const location = useLocation();
  const [path, setPath] = useState();
  useEffect(() => {
    if (location.pathname === '/home') {
      setPath('Home Page');
    } else {
      let path = location.pathname.split('/')[2];
      path = path.split('-').join(' ');
      setPath(path);
    }
  }, [location.pathname]);

  return (
    <div className="navbar">
      <div className="nav-items">
        {location.pathname !== '/home'
          && (
            <NavLink to="/home">
              <img src={backIcon} alt="" aria-label="back" width="30px" />
            </NavLink>
          )}
      </div>
      <h1 className="page-title">{path}</h1>
    </div>
  );
}

export default Navigation;
