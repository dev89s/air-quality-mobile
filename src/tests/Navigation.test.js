import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

describe('Navbar header test', () => {
  it('should show Current Weather', () => {
    render(
      <MemoryRouter initialEntries={[{ pathname: '/home' }]}>
        <Navigation />
      </MemoryRouter>,
    );
    const title = screen.queryByRole('heading');
    expect(title.textContent).toBe('Current Weather');
  });

  it('should show New York', () => {
    render(
      <MemoryRouter initialEntries={[{ pathname: '/city/New-York' }]}>
        <Navigation />
      </MemoryRouter>,
    );
    const title = screen.queryByRole('heading');
    expect(title.textContent).toBe('City/New York');
  });
});

describe('Navbar header test', () => {
  it('should stay the same', () => {
    const nav = renderer.create(
      <MemoryRouter initialEntries={[{ pathname: '/home' }]}>
        <Navigation />
      </MemoryRouter>,
    );
    const tree = nav.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
