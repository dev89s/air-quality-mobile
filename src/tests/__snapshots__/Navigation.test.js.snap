// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar header test should stay the same 1`] = `
<div
  className="navbar"
>
  <h1
    className="page-title"
  />
</div>
`;
